lib/rspec/core/formatters/profile_formatter.rb

```diff
@@ -6,12 +6,29 @@ module Formatters
       # @api private
       # Formatter for providing profile output.
       class ProfileFormatter
-        Formatters.register self, :dump_profile
+        Formatters.register self, :dump_profile, :example_group_started, :example_group_finished, :example_started
 
         def initialize(output)
+          @example_groups = {} #todo rename, maybe groups_data, groups_information or profile_information
           @output = output
         end
 
+        def example_group_started(notification)
+          @example_groups[notification.group.id] = Hash.new(0)
+          @example_groups[notification.group.id][:start] = Time.now
+          @example_groups[notification.group.id][:description] = notification.group.top_level_description
+        end
+
+        def example_group_finished(notification)
+          @example_groups[notification.group.id][:total_time] =  Time.now - @example_groups[notification.group.id][:start]
```

Comment:
The logic you've added to these methods should be moved elsewhere, either to the notifications or to the reporter that generates them so that we're not repeating the logic (and so eventually other formatters can make use of it if they want to)